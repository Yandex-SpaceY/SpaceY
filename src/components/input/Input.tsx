import React from 'react';
import './input.scss';
interface IInput {
  onChange(e: React.ChangeEvent<HTMLInputElement>): void;
  name: string;
  title?: string;
  type?: string;
  errorText?: string;
}

const Input = (props: IInput): React.ReactElement => {
  const { onChange, name, type = 'text', title = name, errorText = '' } = props;

  return (
    <div className="one-input-block">
      <span className="title">{title}</span>
      <input className="input" onChange={onChange} name={name} type={type} />
      <span className="error-message">{errorText}</span>
    </div>
  );
};

export default Input;
