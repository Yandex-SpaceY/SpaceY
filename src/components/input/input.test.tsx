import * as React from 'react';
import { shallow } from 'enzyme';

import Input from './Input';

it('check render component Input', () => {
  const component = shallow(<Input
    value='testValue'
    name='testName'
    title='testTitle'
  />);

  expect(component.text()).toEqual('testTitle');

  expect(component).toMatchSnapshot();
});
