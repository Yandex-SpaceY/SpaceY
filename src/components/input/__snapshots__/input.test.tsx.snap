// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check render component Input 1`] = `
<div
  className="one-input-block"
>
  <label
    className="title"
    htmlFor="testName"
  >
    testTitle
  </label>
  <input
    className="input"
    disabled={true}
    id="testName"
    name="testName"
    type="text"
    value="testValue"
  />
  <span
    className="error-message"
  />
</div>
`;
